interp/run.go

```diff
@@ -1985,7 +1923,28 @@ func getMethodByName(n *node) {
 	l := n.level
 
 	n.exec = func(f *frame) bltn {
-		val := value0(f).Interface().(valueInterface)
+		// The interface object must be directly accessible, or embedded in a struct (exported anonymous field).
+		val0 := value0(f)
+		val, ok := value0(f).Interface().(valueInterface)
+		if !ok {
+			// Search the first embedded valueInterface.
+			for val0.Kind() == reflect.Ptr {
+				val0 = val0.Elem()
+			}
+			for i := 0; i < val0.NumField(); i++ {
+				fld := val0.Type().Field(i)
+				if !fld.Anonymous || !fld.IsExported() {
+					continue
+				}
+				if val, ok = val0.Field(i).Interface().(valueInterface); ok {
+					break
```

Comment:
```suggestion
					break
					// TODO: should we keep track of all the the vals that are indeed valueInterface,
					// so that later on we can call MethodByName on all of them until one matches?
```